pkg/providers/snow/validator.go

```diff
@@ -12,23 +12,39 @@ const (
 	defaultAwsSshKeyName = "eksa-default"
 )
 
-type Validator struct {
+type validator struct {
 	awsClientMap AwsClientMap
 }
 
-func NewValidator(aws aws.Clients) *Validator {
-	return &Validator{
+type Validator interface {
+	ValidateEC2SshKeyNameExists(ctx context.Context, m *v1alpha1.SnowMachineConfig) error
+	ValidateEC2ImageExistsOnDevice(ctx context.Context, m *v1alpha1.SnowMachineConfig) error
+	ValidateMachineDeviceIPs(ctx context.Context, m *v1alpha1.SnowMachineConfig) error
+}
+
+type validatorBuilder struct{}
+
+func NewValidatorBuilder() *validatorBuilder {
+	return &validatorBuilder{}
+}
+
+func (b *validatorBuilder) Build(aws aws.Clients) Validator {
+	return NewValidator(aws)
+}
+
+func NewValidator(aws aws.Clients) Validator {
+	return &validator{
 		awsClientMap: NewAwsClientMap(aws),
 	}
 }
 
-func NewValidatorFromAwsClientMap(awsClientMap AwsClientMap) *Validator {
-	return &Validator{
+func NewValidatorFromAwsClientMap(awsClientMap AwsClientMap) Validator {
```

Comment:
I think this should still return the struct
We should try to "return structs, receive interfaces" when possible (except for the builder)